File: backend/model/entity/stock.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Stock struct {
	gorm.Model
	VPSID         uint `gorm:"uniqueIndex;not null"`
	Status        int  `gorm:"not null;default:3"`
	Quantity      *int
	LastCheckedAt *time.Time
	LastInStockAt *time.Time
	DeliveryID    string `gorm:"size:64;not null" json:"-"`
}

func (Stock) TableName() string { return "vps_stocks" }

File: frontend/src/types/stock.ts
// 库存相关类型契约
import { ID, NullableTimestamp } from './common';

export type StockStatus = 1 | 2 | 3; // 1: 有货, 2: 缺货, 3: 未知

export interface Stock {
  vps_id: ID;
  status: StockStatus;
  quantity: number | null;
  last_checked_at: NullableTimestamp;
  last_in_stock_at: NullableTimestamp;
  is_stale: boolean;
}
